test/unit/puppetlabs/puppetserver/lockable_pool_test.clj

```diff
@@ -235,6 +235,11 @@
       @blocked-borrow-thread-started?
       @lock-thread-started?
       (is (not (realized? blocked-borrow-thread-borrowed?)))
+      (let [start (System/currentTimeMillis)]
+        (while (and (not (.isLocked pool))
+                    (< (- (System/currentTimeMillis) start) 10000))
+          (Thread/yield)))
+      (is (.isLocked pool))
       (is (not (realized? lock-acquired?)))
 
       (return-instances pool instances)
```

Comment:
I was able to reproduce the test failure on the following line...

~~~clj
(is (not (realized? blocked-borrow-thread-borrowed?)))
~~~

... by adding a `Thread.sleep(5000)` between the `(deliver lock-thread-started? true)` and `(.lock pool)` lines in the lock-thread future.

I added the `.isLocked pool` check thinking that `.isLocked` is supposed to return true as soon as the request for the lock is granted (even though this can precede when all of the borrowed instances have been checked back into the pool).  With the code above in place, we're much more likely to have gotten past the critical section of code where the requested lock should hold off borrow attempts - which is what this test is primarily about, I think.

I put a 10 second cap on waiting for the lock to be requested because I didn't want to run into the possibility of a wait hang causing the test run to fail mysteriously.  This roughly follows the pattern elsewhere in these tests where we use `timed-deref`s.  Still racy obviously - which isn't great - but I think the above code should help with the test as is.

We had a discussion at one time about changing the name of `.isLocked` to something else because of concerns about whether this is really fulfilling the definition of "locked" in that it returns true between the phases of "caller has been established as the next lock holder" and "currently borrowed instances have been checked back into the pool".  Still open to changing that name at some point but probably outside of the scope of this PR.